frontend/src/pages/Home/Home.tsx

```diff
@@ -1,24 +1,79 @@
-import { FloatButton, Layout, Typography } from 'antd'
+import { Button, FloatButton, Layout, List, Typography } from 'antd'
+import { useItems } from 'services/item_service'
 import React from 'react'
 import { Link, Outlet } from 'react-router-dom'
 
+import { useActivities, useUpdateActivity } from 'services/activities_service'
+import { usePagedResultData } from 'services/utils'
+
 export interface HomeProps {}
 
 const Home = ({}: HomeProps) => {
+
+    const { data: recentsPagedResult, isPending: recentsPending, fetchStatus: recentsStatus } = useItems(1, 5, undefined, undefined, {key: "created", order: "descend"})
+    const { data: recents} = usePagedResultData(recentsPagedResult)
+
+    const { data: upcomingPagedResult, isPending: upcomingPending, fetchStatus: upcomingStatus } = useActivities(1, 5, undefined, {completed:['false'], pending:['true']})
+    const { data: upcoming } = usePagedResultData(upcomingPagedResult)
+
+    const { data: unfinishedPagedResult, isPending: unfinishedPending, fetchStatus: unfinishedStatus } = useActivities(1, 5, undefined, {completed:['false'], pending:['false'], hasEndTime:['false']})
+    const { data: unfinished } = usePagedResultData(unfinishedPagedResult)
+
+    const activityUpdateMutation = useUpdateActivity()
+
+    const markFinished = (item:any) => {
```

Comment:
arg should be activity and the type should be Activity, yea?

Reply:
probably. think I meant to return to this to add proper type.